### PR TITLE
TST/CLN: Use fixtures instead of setup_method in tests/indexing

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/tests/indexing/common.py [hunk 1]
```diff
@@ -1,190 +1,28 @@
...
-
-        self.frame_empty = DataFrame()
-        self.series_empty = Series(dtype=object)
-
-        # form agglomerates
-        for kind in self._kinds:
-            d = {}
-            for typ in self._typs:
-                d[typ] = getattr(self, f"{kind}_{typ}")
+def check_result(obj, method, key, axes=None, fails=None):
```
Comment (jreback) [inline pandas/tests/indexing/common.py:8]: worth typing the args?

File: pandas/tests/indexing/test_iloc.py [hunk 2]
```diff
@@ -32,7 +32,7 @@
 )
 import pandas._testing as tm
 from pandas.api.types import is_scalar
-from pandas.tests.indexing.common import Base
+from pandas.tests.indexing.common import check_result
```
Comment (jreback) [inline pandas/tests/indexing/test_iloc.py:35]: maybe rename to assert_result (or something slightly more verbose)

File: pandas/tests/indexing/test_loc.py [hunk 3]
```diff
@@ -107,25 +116,31 @@ def test_loc_getitem_label_list(self, key, typs):
             [[(1, 3), (1, 4), (2, 5)], ["multi"], 0],
         ],
     )
-    def test_loc_getitem_label_list_with_missing(self, key, typs, axes):
-        self.check_result("loc", key, typs=typs, axes=axes, fails=KeyError)
-
-    def test_loc_getitem_label_list_fails(self):
+    @pytest.mark.parametrize("kind", ["series", "frame"])
```
Comment (jreback) [inline pandas/tests/indexing/test_loc.py:119]: fixture for the kind?

**Review by jreback — CHANGES_REQUESTED** [review]
a few comments

File: pandas/tests/indexing/common.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/indexing/common.py]: Sure, typed the straightforward args

File: pandas/tests/indexing/test_iloc.py (hunk 2 above)
Comment (mroeschke) [reply pandas/tests/indexing/test_iloc.py]: Renamed to `check_indexing_smoketest_or_raises`

File: pandas/tests/indexing/test_loc.py (hunk 3 above)
Comment (mroeschke) [reply pandas/tests/indexing/test_loc.py:121]: Don't think it's entirely worth it here. This is just used to combine with another string to fetch a fixture.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**